File: 2427-number-of-common-factors/2427-number-of-common-factors.js
/**
 * @param {number} a
 * @param {number} b
 * @return {number}
 */
var commonFactors = function(a, b) {
    let cnt = 0;
    let n = (Math.min(a, b) + 1);
    
    for (let i = 0; i <= n; i++){
        if (a % i === 0 && b % i === 0) {
            cnt++;
        }
    }
    return cnt;
};

File: 2427-number-of-common-factors/2427-number-of-common-factors.cpp
class Solution {
public:
    int commonFactors(int a, int b) {
        int cnt = 0;
        int n = min(a, b) + 1;

        for (int i = 1; i <= n; i++) {
            if (a % i == 0 && b % i == 0) {
                cnt++;
            }
        }
        return cnt;
    }
};
